helpdesk/management/commands/get_email.py

```diff
@@ -234,17 +234,17 @@ def process_queue(q, logger):
         logger.info("Found %s messages in local mailbox directory" % str(len(mail)))
         for m in mail:
             logger.info("Processing message %s" % str(m))
-            f = open(m, 'r')
-            ticket = ticket_from_message(message=f.read(), queue=q, logger=logger)
-            if ticket:
-                logger.info("Successfully processed message %s, ticket/comment created." % str(m))
-                try:
-                    # unlink(m) #delete message file if ticket was successful
-                    logger.info("Successfully deleted message %s." % str(m))
-                except:
-                    logger.error("Unable to delete message %s." % str(m))
-            else:
-                logger.warn("Message %s was not successfully processed, and will be left in local directory" % str(m))
+            with open(m, 'r') as f:
+                ticket = ticket_from_message(message=f.read(), queue=q, logger=logger)
+                if ticket:
+                    logger.info("Successfully processed message %s, ticket/comment created." % str(m))
+                    try:
+                        # unlink(m) #delete message file if ticket was successful
```

Comment:
Yeah, that's how I see it. This behaviour is analgous to how we deal with POP3/IMAP mail, it's just that we don't have the extra protocol layer to pass deletions through since this mail is delivered directly as file-on-disk IIRC. There are past issues about duplicate ticket creation by emails that didn't get deleted, e.g. due to exceptions thrown during attachment processing so I think that loud and explicit documentation remains the best option. $0.02...